### PR TITLE
BOPTEST-service update

### PR DESCRIPTION
**Pull request by ThibaultMarzullo**
This PR is to update the ACTB with the new changes to boptest-service and Spawn.
BOPTEST-service updates:
- boptest-service now uses the Alfalfa docker containers, and cloud-based services,
- all test cases are submitted to the simulation worker, not just a single one at a time,
- the RESTful API has been updated to reflect the presence of several test cases at once,
- the test case resources have been reorganized,
- simulation is MUCH faster.

ACTB updates:
- new client that reflects the changes to the RESTful API,
- all interfaces have been updated to work with the new client,
- all examples have been updatet to work with the new client,
- the Spawn test case has been updated
- the simulation worker now contains resources for running Spawn test cases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,16 +51,18 @@ are installing these libraries for your Python 3.
 ## Quick-Start to Run Test Cases
 
 1. Make sure the Docker daemon is running. 
-2. Build the Spawn Small Office test case by running ``$ make build TESTCASE=spawnrefsmalloffice``
-3. Run the Spawn Small Office test case by running ``$ make run TESTCASE=spawnrefsmalloffice``
+2. Build the Docker containers by running ``$ make build``
```
Comment (nllong) [inline README.md:54]: This errors with
```
docker build --build-arg testcase= --no-cache --rm -t boptest_ .
invalid argument "boptest_" for "-t, --tag" flag: invalid reference format
```
We still need to pass in the TESTCASE to use the makefile

Comment (nllong) [reply README.md:54]:
```suggestion
2. Build the Docker containers by running ``$ make build TESTCASE=spawnrefsmalloffice``
```

File: README.md [hunk 2]
```diff
@@ -51,16 +51,18 @@ are installing these libraries for your Python 3.
...
+3. Run the Docker containers by running ``$ make run``
```
Comment (nllong) [inline README.md:55]:
```suggestion
3. Run the Docker containers by running ``$ make run TESTCASE=spawnrefsmalloffice``
```

File: README.md [hunk 3]
```diff
@@ -51,16 +51,18 @@ are installing these libraries for your Python 3.
...
 4. Run an example test controller in a separate terminal:
 
   * ``$ cd examples/python/MPC-spawnrefsmalloffice && python main.py`` to test a MPC controller.
   * ``$ cd examples/python/RLC-spawnrefsmalloffice && python DQN_Test_run.py`` to test a RLC controller.
   * ``$ cd examples/python/RBC-spawnrefsmalloffice && python supervisory_example.py`` to test the RBC controller.
  
-4. Shutdown a test case container by selecting the container terminal window, ``Ctrl+C`` to close port, and ``Ctrl+D`` to exit the Docker container.
-5. Remove the test case Docker image by ``$ make remove-image TESTCASE=spawnrefsmalloffice``.
+4. Shutdown a test case container by selecting the container terminal window, ``Ctrl+C`` to close port, and ``make stop`` to stop the Docker container.
+5. Remove the test case Docker image by ``$ make remove-image``.
```
Comment (nllong) [inline README.md:63]:
```suggestion
5. Remove the test case Docker image by ``$ make remove-image TESTCASE=spawnrefsmalloffice``.
```

File: examples/python/MPC-spawnrefsmalloffice/main.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 from template_mpc import template_mpc
```
Comment (nllong) [inline examples/python/MPC-spawnrefsmalloffice/main.py:17]: I fixed the plotting in this PR. Hopefully this was an issue for you too.

https://github.com/henze-research-group/MODRLC/pull/23

**Review by nllong — APPROVED** [review]
nice cleanup and merging! Few comments and one [PR](https://github.com/henze-research-group/MODRLC/pull/23)

File: README.md (hunk 1 above)
Comment (ThibaultMarzullo) [reply README.md:54]: The new version of BOPTEST-service uses a new makefile that uses docker-compose, this works on my end.
This is the new makefile on the ```boptest-service``` branch:

```
build:
	docker-compose build

remove-image:
	docker-compose rm -sf

run :
	$(MAKE) run-detached
	$(MAKE) provision
	docker-compose logs -f web worker

bash:
	$(COMMAND_RUN) --detach=false ${IMG_NAME} /bin/bash -c "bash"

run-detached:
	docker-compose up -d web worker

provision:
	docker-compose run --no-deps provision python3 -m boptest_submit ./testcases/${TESTCASE}

stop:
	docker-compose down

.PHONY: build run run-detached remove-image stop provision
```

File: README.md (hunk 2 above)
Comment (ThibaultMarzullo) [reply README.md:55]: Same as [this comment](https://github.com/henze-research-group/MODRLC/pull/22#discussion_r761618542)

File: README.md (hunk 3 above)
Comment (ThibaultMarzullo) [reply README.md:63]: Same as [this comment](https://github.com/henze-research-group/MODRLC/pull/22#discussion_r761618542)

File: examples/python/MPC-spawnrefsmalloffice/main.py (hunk 4 above)
Comment (ThibaultMarzullo) [reply examples/python/MPC-spawnrefsmalloffice/main.py:17]: I see an issue with the last tick. Hour 24 (or rather 0) does not show. I will add a fix to your PR to show the last tick.

**Pull request merged**